tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -85,79 +103,108 @@ static_assert(
                              VariablesTestTags_detail::vector>>>::value,
     "Missing move semantics in Variables.");
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables", "[DataStructures][Unit]") {
-  Variables<tmpl::list<VariablesTestTags_detail::vector,
-                       VariablesTestTags_detail::scalar,
-                       VariablesTestTags_detail::scalar2>>
-      v(1, -3.0);
-  decltype(v) v_initialize;
-  v_initialize.initialize(1, -3.0);
-  CHECK(v == v_initialize);
-
-  CHECK(v.size() ==
-        v.number_of_grid_points() * v.number_of_independent_components);
-
-  CHECK(1 == v.number_of_grid_points());
-  CHECK(5 == v.size());
-
-  auto& vector_in_v = get<VariablesTestTags_detail::vector>(v);
+template <typename VectorType>
+void test_variables_construction_and_access() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<
+      typename tt::get_fundamental_type_t<typename VectorType::value_type>>
+      dist{-100, 100};
+
+  size_t number_of_grid_points = 1;
+  auto initial_fill_value =
+      make_with_random_values<typename VectorType::value_type>(
+          make_not_null(&gen), make_not_null(&dist));
+
+  Variables<tmpl::list<VariablesTestTags_detail::vector_tag<VectorType>,
+                       VariablesTestTags_detail::scalar_tag<VectorType>,
+                       VariablesTestTags_detail::scalar2_tag<VectorType>>>
+      filled_variables{number_of_grid_points, initial_fill_value};
+
+  decltype(filled_variables) initialized_variables;
+  initialized_variables.initialize(number_of_grid_points, initial_fill_value);
+  CHECK(filled_variables == initialized_variables);
+
+  CHECK(filled_variables.size() ==
+        filled_variables.number_of_grid_points() *
+            filled_variables.number_of_independent_components);
+  CHECK(number_of_grid_points == filled_variables.number_of_grid_points());
+  CHECK(number_of_grid_points * 5 == filled_variables.size());
+
+  auto& vector_in_filled_variables =
```

Comment:
Please add a comment `// Test that assigning to a tensor pointing into a Variables correctly sets the Variables`